### PR TITLE
Bugfix/marketplace fixes

### LINKED ISSUE #164: Marketplace Sales don't delete themselves after finished.
**Describe the bug**
Sales don't delete themselves from DB when done being used

**To Reproduce**
1. Log on and go to market
2. put something for sale
3. log on someone else
4. buy the sale
5. sale doesn't vanish

**Expected behavior**
Sale should delete itself

**Screenshots**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Derplime**
**Description**
A few things were done here:
- Changed the API for getMarketplaceSales to accept a parameter containing the userId. This then filters the DB search to sales that are not associated with the userId. The frontend is updated as well. @Zexdur a race case does happen for this, talk to me when you get a chance.
- Sales are now deleted upon completed transactions. Last time I checked, the list doesn't refresh to reflect the deleted sale.
- Unrelated update: npm crypto package is old (like 3 years old), so removed it from package.json since Nodejs has its own crypto library built in. It works the same with the current implementation.

Resolves #164 

**Type of change**
Check options that are relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Fix or feature that would cause existing functionality to not work as expected
- [ ] This change requires a update in the design document

**Steps to Verify Functionality**
Make a bulleted/numbered list of steps to verify this feature/bugfix. Include screenshots if necessary.

**Final Checklist:**
Go down the list and check them off.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the design document (if applicable)
- [x] My changes generate no new warnings
- [x] Pulled updated master branch (if changed since branch creation) and corrected any conflicts, if any occur.

### CODE REVIEW

File: src/backend/controllers/marketController.js [hunk 1]
```diff
@@ -151,7 +151,7 @@ exports.addMarketSale = async (req: $Request, res: $Response) => {
                 await sale.save();
 
                 // Send back success confirmation
-                res.status(201).json({ msg: 'Successfully created Market Sale.' });                    
+                res.status(201).json({ msg: 'Successfully created Casus Block Market Sale.' });
```
Comment (SecondThread) [inline src/backend/controllers/marketController.js]: hmmm, quite possible we just don't get to this :(

File: src/backend/controllers/marketController.js [hunk 2]
```diff
@@ -174,6 +195,7 @@ exports.getMarketSales = async (req: $Request, res: $Response) => {
 
         // Return list of sales
         res.status(200).json(salesList);
```
Comment (SecondThread) [inline src/backend/controllers/marketController.js]: Do we not need to return this?

File: src/backend/controllers/marketController.js [hunk 3]
```diff
@@ -195,6 +217,7 @@ exports.getMarketSale = async (req: $Request, res: $Response) => {
 
         // Return sale confirmation
         res.status(200).json(sale);
```
Comment (SecondThread) [inline src/backend/controllers/marketController.js]: same thing, do we not need a return keyword?

**Review by SecondThread — APPROVED** [review]

File: src/backend/controllers/marketController.js (hunk 1 above)
Comment (Derplime) [reply src/backend/controllers/marketController.js]: Yeah but might as well have it there for when you're able to. Unless that might result in a problem.

**Comment by JorgeVidal29** [comment]
Looks good

**Review by JorgeVidal29 — APPROVED** [review]

**Pull request merged**